packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2742,6 +2761,10 @@ class InputDecoration {
          !(label != null && labelText != null),
          'Declaring both label and labelText is not supported.',
        ),
+       assert(
+         !(hintText != null && hint != null),
+         'Declaring both hintText and hint is not supported.',
```

Comment:
Very minor nit: to make this assert message similar to the other checks consider changing the words order "both hint and hintText".

Reply:
updated the assert message

Reply:
Added tests